src/contracts/CanonGuard.sol

```
@@ -146,7 +146,9 @@ contract CanonGuard is OnlyCanonGuard, EmergencyModeHook, ICanonGuard {
   }
 
   /// @inheritdoc ICanonGuard
-  function executeTransaction(address _actionsBuilder) external payable {
+  function executeTransaction(address _actionsBuilder) public payable {
+    _onBeforeExecution();
```

I moved this on top to fail faster in case it is in emergency mode. wdyt? I think in the batch function makes a bit more sense so wanted to keep it consistent

---

Perfect!